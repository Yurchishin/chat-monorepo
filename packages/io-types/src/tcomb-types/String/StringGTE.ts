import t from 'tcomb-validation'
import { StringGT } from './StringGT'
import { StringEqualLength } from './StringEqualLength'

export const StringGTE = (min: number) => t.union([
    StringGT(min),
    StringEqualLength(min),
], 'StringGTE')
