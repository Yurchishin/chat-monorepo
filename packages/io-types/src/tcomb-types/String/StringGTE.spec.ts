import {StringGTE} from './StringGTE'
import {expectTypeSuccess, expectTypeFailure} from "../../tests"

describe('StringGTE(5)', () => {
	const GTE5 = StringGTE(5)

	expectTypeSuccess(GTE5, '666666')
	expectTypeSuccess(GTE5, '55555')
	expectTypeFailure(GTE5, '4444')
	expectTypeFailure(GTE5, 1)
})
