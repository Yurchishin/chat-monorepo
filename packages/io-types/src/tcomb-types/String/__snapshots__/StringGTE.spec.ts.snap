// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StringGTE(5) expectTypeFailure(1): TYPE_FAILURE::StringGTE(1) 1`] = `
Struct {
  "errors": Array [
    Struct {
      "actual": 1,
      "expected": [Function],
      "message": "Invalid value 1 supplied to StringGTE",
      "path": Array [],
    },
  ],
  "value": 1,
}
`;

exports[`StringGTE(5) expectTypeFailure(4444): TYPE_FAILURE::StringGTE(4444) 1`] = `
Struct {
  "errors": Array [
    Struct {
      "actual": "4444",
      "expected": [Function],
      "message": "Invalid value \\"4444\\" supplied to StringGTE",
      "path": Array [],
    },
  ],
  "value": "4444",
}
`;
